Formula/scorecard.rb

```diff
@@ -0,0 +1,28 @@
+class Scorecard < Formula
+  desc "Security health metrics for Open Source"
+  homepage "https://github.com/ossf/scorecard"
+  url "https://github.com/ossf/scorecard/archive/refs/tags/v2.0.0.tar.gz"
+  sha256 "9d3f0dd0a82cdc4cb3cd69149ad0967291b1bafb751bb07bc9c92db00af613a9"
+  license "Apache-2.0"
+  head "https://github.com/ossf/scorecard.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    system "make", "build-scorecard"
+    bin.install "scorecard"
```

Comment:
@owenvoke You wrote (https://github.com/Homebrew/homebrew-core/pull/80441#discussion_r663087178):
> I'd normally have just used `make build`, but that seems to try and install it with a Docker environment. So I've just used the `build-scorecard` which doesn't actually install it into the system (hence why the `bin.install` is needed to add it to the path).

What `build-scorecard` actually does is (https://github.com/ossf/scorecard/blob/v2.0.0/Makefile#L96-L98):
```make
build-scorecard: ## Runs go build on repo
	# Run go build and generate scorecard executable
	CGO_ENABLED=0 go build -a -tags netgo -ldflags '-w -extldflags "-static"'
```
Tags aren't even used through codebase so `-tags netgo` doesn't affect build in any way and `-extldflags "static"` could also be omitted because no dynamic libraries are being used in the first place. i.e. This can be simplified to:
```sh
go build -ldflags '-s -w'
```
_(I added `-s` too because stripping DWARF symbol table (`-w`) without stripping symbol table and debug info (`-s`) doesn't make much sense.)_

in which case you can simply leverage [`std_go_args` builtin](https://rubydoc.brew.sh/Formula.html#std_go_args-instance_method):
```rb
system "go", "build", *std_go_args(ldflags: "-s -w")
``` 
as it's been done inside many other go formulas. Note that using `std_go_args` places build artifact directly inside `bin` so `bin.install ...` is no longer needed.

Additionally, it would be nice to (re)generate `checks.md` and store that to `doc` (taken from `generate-docs` makefile target (https://github.com/ossf/scorecard/blob/v2.0.0/Makefile#L90-L94)):
```rb
cd("checks/main") { system "go", "run", "main.go" }
doc.install "checks/checks.md"
```

So finally:

```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w")
    cd("checks/main") { system "go", "run", "main.go" }
    doc.install "checks/checks.md"
```

Reply:
Thank you for the explanation, and for providing the necessary formatting. :+1: I've not contributed any Go formulae before, so hadn't seen the `std_go_args()` function. Cheers.